### ISSUE TITLE
[Bug]: equinix_fabric_service_profiles is not working for GCP (AWS and Azure are OK!)

### ISSUE THREAD

**Issue by mmartinov-equinix** [equinix/terraform-provider-equinix#596]
### Terraform Version

Terraform v1.7.2

### Equinix Provider Version

equinix/equinix v1.30.0

### Effected Terraform Resources

`equinix_fabric_service_profiles`

### Expected Behavior

Should discover Service Profile, so I can use it in next step with `equinix_fabric_connection` for z_side UUID.

### Actual Behavior
```
Error: Plugin did not respond
  with data.equinix_fabric_service_profiles.gcp,
  on fabric.tf line 1, in data "equinix_fabric_service_profiles" "gcp":
   1: data "equinix_fabric_service_profiles" "gcp" {
 The plugin encountered an error, and failed to respond to the
 plugin.(*GRPCProvider).ReadDataSource call. The plugin logs may contain
 more details.
...
...
Error: The terraform-provider-equinix_v1.30.0.exe plugin crashed!
```
### Steps to Reproduce

Try 
```hcl
data "equinix_fabric_service_profiles" "gcp" {
  filter {
    property = "/name"
    operator = "="
    values   = ["Google Cloud Partner Interconnect Zone 1"]
  }
}
```

with required authentication.

**Comment by ocobles** [comment]
After a quick test it seems the GCP service profile is producing a panic error in https://github.com/equinix/terraform-provider-equinix/blob/12c5b6b21a6d385cca5863d6c58a5a3702b6a012/equinix/resource_fabric_service_profile.go#L909

`panic: interface conversion: interface {} is []map[string]interface {}, not []interface {}
`

Value returned from the API

```
"marketingInfo": {
2024-02-28T14:45:38.561+0100 [DEBUG] provider.terraform-provider-equinix:     "logo": "google_cloud.svg",
2024-02-28T14:45:38.561+0100 [DEBUG] provider.terraform-provider-equinix:     "promotion": true,
2024-02-28T14:45:38.561+0100 [DEBUG] provider.terraform-provider-equinix:     "processSteps": [
2024-02-28T14:45:38.561+0100 [DEBUG] provider.terraform-provider-equinix:      {
2024-02-28T14:45:38.561+0100 [DEBUG] provider.terraform-provider-equinix:       "title": "Create a VLAN attachment in Google Cloud Console",
2024-02-28T14:45:38.561+0100 [DEBUG] provider.terraform-provider-equinix:       "subTitle": "",
2024-02-28T14:45:38.561+0100 [DEBUG] provider.terraform-provider-equinix:       "description": "<p>Create a VLAN attachment for a Partner Interconnect.</p>\n<p>For better performance, create your VLAN attachment in the region closer to your Google Cloud VM and resources. For example, if your VMs are deployed in us-central1, create your VLAN attachment in us-central1.</p>\n<p>This step generates a pairing key that you will share with Equinix. The pairing key is a unique key that allows Equinix to identify and connect to your VPC network and associated Cloud Router. Pairing keys use the following format: &lt;random&gt;/&lt;vlan-attachment-region&gt;/&lt;edge-availability-domain&gt;.</p>\n<p>For example, 7e51371e-72a3-40b5-b844-2e3efefaee59/us-central1/2 is a pairing key for a VLAN attachment in the us-central1 region and edge availability domain 2</p>"
2024-02-28T14:45:38.561+0100 [DEBUG] provider.terraform-provider-equinix:      },
2024-02-28T14:45:38.561+0100 [DEBUG] provider.terraform-provider-equinix:      {
2024-02-28T14:45:38.561+0100 [DEBUG] provider.terraform-provider-equinix:       "title": "Create Connection on the Equinix Fabric",
2024-02-28T14:45:38.561+0100 [DEBUG] provider.terraform-provider-equinix:       "subTitle": "",
2024-02-28T14:45:38.561+0100 [DEBUG] provider.terraform-provider-equinix:       "description": "<p>Complete the primary or secondary connection workflow on the Equinix Fabric by selecting the appropriate Google Cloud edge availability domain - Zone 1. Select Create Connection to Google Cloud Platform below to get started.</p>\n<p>Select the destination of your virtual connection. Each destination can serve several GCP regions. The region shown in the button is the closest but other can be reached from that destination. Please refer to the public <a href=\"https://cloud.google.com/network-connectivity/docs/interconnect/concepts/choosing-colocation-facilities\" target=\"_blank\" rel=\"nofollow noopener noreferrer\">Google's Locations table</a> to verify that your selected destination can serve the region of your VLAN attachment.</p>\n<p>For example, in the table is shown that from the destination Ashburn, all GCP regions in North America are reachable. Another example. from the destination Stockholm, all the GCP regions in Europe can be reached.</p>\n<p>As a best practice, create the VLAN attachment in the GCP region closer to your GCP resources, and</p>\n<p>choose the location closest to your resources in Equinix as destination or the Equinix Fabric connection.</p>"
2024-02-28T14:45:38.561+0100 [DEBUG] provider.terraform-provider-equinix:      },
2024-02-28T14:45:38.561+0100 [DEBUG] provider.terraform-provider-equinix:      {
2024-02-28T14:45:38.561+0100 [DEBUG] provider.terraform-provider-equinix:       "title": "Activate VLAN attachment in Google Cloud Console",
2024-02-28T14:45:38.561+0100 [DEBUG] provider.terraform-provider-equinix:       "subTitle": "",
2024-02-28T14:45:38.561+0100 [DEBUG] provider.terraform-provider-equinix:       "description": "<p>Activate your GCI Partner Interconnect VLAN attachment in the GCP Management Console.</p>"
2024-02-28T14:45:38.561+0100 [DEBUG] provider.terraform-provider-equinix:      }
2024-02-28T14:45:38.561+0100 [DEBUG] provider.terraform-provider-equinix:     ]
2024-02-28T14:45:38.561+0100 [DEBUG] provider.terraform-provider-equinix:    },
```

**Comment by thogarty** [comment]
Hi @mmartinov-equinix, this bug was detected by us earlier and we have already implemented the fix and it is available in version [1.31.0](https://registry.terraform.io/providers/equinix/equinix/latest/docs) of the provider. 

Bug fix implemented in https://github.com/equinix/terraform-provider-equinix/pull/582

Closing issue based on bug fix released.
